### DISCUSSION TITLE
Multiple instances of tmkms-light with SGX provider

### DISCUSSION THREAD

**Discussion by tomtau** [crypto-com/tmkms-light#462] [Q&A] [answered]
> Hello,
> How many instances of tmkms-light with an SGX provider could be run on a single machine with Total EPC size: 93.5MiB?
> Many thanks for considering my request.

Originally posted by @ekhvalov in https://github.com/tomtau/tmkms-light/issues/66#issue-1449332772

**Comment by tomtau** [comment, marked as answer]
AFAIK on linux, SGX memory can be swapped in from and out to main memory (while being encrypted/decrypted), so it's not limited by EPC, but it may become slower with more instances being loaded and swapped out frequently. I'd say 2-3 may be possible, but perhaps still better to have a machine dedicated to a single signer.

**Reply by ekhvalov** [reply]
Thank you so much.
